src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -618,6 +619,41 @@ public override BoundNode VisitRefTypeOperator(BoundRefTypeOperator node)
             return node.Update(operand, getTypeFromHandle, type);
         }
 
+        private BoundStatement? RewriteFieldOrPropertyInitializer(BoundStatement initializer)
+        {
+            var previousLocals = _additionalLocals;
```

Comment:
I didn't understand, why introduce this new `_additionalLocals` machinery, which affects every single block, when one can easily insert a new block during lowering of specific nodes when additional locals are needed?

Reply:
The inline array temporary needs to be added in the enclosing scope (so the lifetime of the temporary matches that scope) rather than being created in a new nested block.

Reply:
I see. Thanks much for the explanation. Consider adding a comment on the declaration of `_additionalLocals` (allows adding locals without inserting new blocks/scopes)